src/main/java/build/buildfarm/instance/shard/RedisShardBackplane.java

```diff
@@ -1479,4 +1490,40 @@ public GetClientStartTimeResult getClientStartTime(String clientKey) throws IOEx
       return GetClientStartTimeResult.newBuilder().build();
     }
   }
+
+  private void startPrometheusMetricsCollector() {
+    prometheusMetricsThread =
+      new Thread(
+        () -> {
+          while (true) {
+            try {
+              TimeUnit.SECONDS.sleep(30);
+              OperationsStatus operationsStatus = operationsStatus();
+              PrometheusPublisher.updateWorkerPoolSize(operationsStatus.getActiveWorkersCount());
+              PrometheusPublisher.updateDispatchedOperationsSize(operationsStatus.getDispatchedSize());
+              PrometheusPublisher.updatePreQueueSize(operationsStatus.getPrequeue().getSize());
+              PrometheusPublisher.updateClusterUtilization();
+              updateQueueSizes(operationsStatus.getOperationQueue().getProvisionsList());
+            } catch (InterruptedException e) {
+              Thread.currentThread().interrupt();
+              break;
+            } catch (Exception e) {
+              logger.log(Level.SEVERE, "Could not update RedisShardBackplane metrics", e);
+            }
+          }
+        },
+        "Prometheus Metrics Collector");
+
+    prometheusMetricsThread.start();
+  }
+
+  private void updateQueueSizes(List<QueueStatus> queues) {
+    for (QueueStatus queueStatus : queues) {
+      if (queueStatus.getName().contains("cpu")) {
+        PrometheusPublisher.updateCpuQueueSize(queueStatus.getSize());
+      } else if (queueStatus.getName().contains("gpu")) {
+        PrometheusPublisher.updateGpuQueueSize(queueStatus.getSize());
```

Comment:
Can this be more generic? The names of the queues are customizable and I would be nice if this were derived instead.  If not, its okay for now.

Reply:
Yeah it needs to be made generic. I just didn't think of a way to do it yet, but wanted to at least have these exposed in the initial run rather than not.